examples/vectorization.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Example showing how to use vectorization of FunctionSpaceVector's."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+import numpy as np
+import odl
+from odl.util.vectorization import vectorize
+import timeit
+
+
+def performace_example():
+    # Create a function space
+    X = odl.FunctionSpace(odl.Interval(0, 1))
+
+    # Functions default to vectorized
+    f_vec = X.element(lambda x: x ** 2)
+
+    # If 'vectorized=False' is used, odl automatically vectorizes
+    f_novec = X.element(lambda x: x ** 2, vectorized=False)
+
+    # Example of runtime, expect vectorized to be much faster
+    points = np.linspace(0, 1, 10000)
+
+    print('Vectorized runtime:     {:5f}'
+          ''.format(timeit.timeit(lambda: f_vec(points), number=100)))
+    print('Non-vectorized runtime: {:5f}'
+          ''.format(timeit.timeit(lambda: f_novec(points), number=100)))
+
+
+def numba_example():
+    # Some functions are not easily vectorized,
+    # here we can use numba to improve performance.
+
+    try:
+        import numba
+    except ImportError:
+        print('Numba not installed, skipping.')
+        return
+
+    def myfunc(x):
+        """Return x - y if x > y, otherwise return x + y."""
+        if x[0] > x[1]:
+            return x[0] - x[1]
+        else:
+            return x[0] + x[1]
+
+    # Numba expects functions f(x1, x2, x3, ...), while we have the
+    # convention f(x) with x = (x1, x2, x3, ...). Therefore we need
+    # to wrap the Numba-vectorized function.
+    vectorized = numba.vectorize(lambda x, y: x - y if x > y else x + y)
+
+    def myfunc_vec(x):
+        """Return x - y if x > y, otherwise return x + y."""
+        return vectorized(*x)
```

Comment:
Perhaps we should write `x[0], x[1]` since this is easier for new users to understand.